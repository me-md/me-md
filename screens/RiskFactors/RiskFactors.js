import React, { useEffect, useState } from 'react';
import { Dimensions, StyleSheet, Text, View } from 'react-native';
import { CheckBox } from 'react-native-elements';
import { LinearGradient } from 'expo-linear-gradient';
import { Body, Card, CardItem, Item, Input } from 'native-base';
import { Entypo } from '@expo/vector-icons';
import { getCommonRiskFactors } from '../../utils/apiCalls/apiCalls';

const height = Dimensions.get('window').height;

export default function RiskFactors({ navigation }) {
  const { age, location, sex } = navigation.state.params;
  const [factors, setFactors] = useState([]);
  const [presentFactors, setPresentFactors] = useState([
    {
      id: 'p_7',
      present: false
    },
    {
      id: 'p_8',
      present: false
    },
    {
      id: 'p_9',
      present: false
    },
    {
      id: 'p_10',
      present: false
    },
    {
      id: 'p_28',
      present: false
    }
  ]);

  const getRiskFactors = async () => {
    const riskFactors = await getCommonRiskFactors();
    setFactors(riskFactors.data);
  };

  const findRiskToUpdate = id => {
    let found = presentFactors.findIndex(factor => factor.id == id);
    setPresentFactors([
      ...presentFactors,
      (presentFactors[found].present = !presentFactors[found].present)
    ]);
  };

  const riskFactorAnswers = factors.map(factor => {
    let foundIndex = presentFactors.findIndex(
      question => factor.id == question.id
    );
    return (
      <Card key={factor.id} style={styles.riskFactorCard}>
        <CardItem>
          <Body>
            <Text style={styles.questionText}>{factor.question}</Text>
            <View style={styles.checkboxes}>
              <CheckBox
                center
                id={factor.id}
                title={<Text>Yes</Text>}
                checked={presentFactors[foundIndex].present}
                onPress={() => {
                  findRiskToUpdate(factor.id);
                }}
              ></CheckBox>
            </View>
          </Body>
        </CardItem>
      </Card>
    );
  });

  useEffect(() => {
    getRiskFactors();
  }, []);

  return (
    <View style={styles.container}>
      <LinearGradient
        style={styles.gradientBackground}
        colors={['#004EFF', '#88CCF1']}
      >
        <View style={styles.screenContainer}>
          <Entypo
            name='chevron-thin-up'
            size={36}
            color='white'
            onPress={() => navigation.navigate('Location')}
          />
          <View style={styles.contentContainer}>
            <Text style={styles.header}>Common Risk Factors:</Text>
            {riskFactorAnswers}
          </View>
          <Entypo
            name='chevron-thin-down'
            size={36}
            color='white'
            onPress={() =>
              navigation.navigate('SearchSymptoms', {
                age,
                location,
                sex,
                present
              })
            }
          />
        </View>
      </LinearGradient>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1
  },
  screenContainer: {
    alignItems: 'center',
    flex: 1,
    justifyContent: 'center'
  },
  gradientBackground: {
    flex: 1
  },
  contentContainer: {
    alignItems: 'center',
    flex: 0.9,
    justifyContent: 'space-evenly',
    marginBottom: height * 0.015,
    marginTop: height * 0.015,
    width: '100%'
  },
  header: {
    color: 'white',
    fontSize: 36,
    fontWeight: 'bold',
    marginBottom: 30,
    marginTop: height * 0.01
  },
  questionText: {
    alignSelf: 'flex-start',
    fontSize: 16
  },
  riskFactorCard: {
    flex: 1,
    marginBottom: height * 0.01,
    marginTop: height * 0.01,
    width: '75%'
  },
  checkboxes: {
    alignSelf: 'flex-start',
    flexDirection: 'row',
    fontSize: 14,
    marginTop: height * 0.01
  }
});
