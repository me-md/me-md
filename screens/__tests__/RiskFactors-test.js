import React from 'react';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import renderer from 'react-test-renderer';

import RiskFactors from '../../screens/RiskFactors/RiskFactors';

jest.mock('expo', () => ({
  AppLoading: 'AppLoading'
}));

describe('RiskFactors', () => {
  jest.useFakeTimers();

  beforeEach(() => {
    NavigationTestUtils.resetInternalState();
  });

  it(`should match a snapshot`, () => {
    const mockNavigation = {
      state: {
        params: {
          sex: 'male',
          age: 24,
          stateAbbreviation: 'CO',
          location: {
            lat: 121,
            long: 25
          }
        }
      }
    };

    const tree = renderer
      .create(<RiskFactors navigation={mockNavigation} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
