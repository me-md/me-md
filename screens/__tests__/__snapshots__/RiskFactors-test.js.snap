// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RiskFactors should match a snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <ViewManagerAdapter_ExpoLinearGradient
    proxiedProperties={
      Object {
        "colors": Array [
          4278210303,
          4287155441,
        ],
        "endPoint": undefined,
        "locations": undefined,
        "startPoint": undefined,
      }
    }
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "space-evenly",
          "marginBottom": 80.03999999999999,
          "marginTop": 80.03999999999999,
        }
      }
    >
      <Text />
      <View
        style={
          Object {
            "alignContent": "space-between",
            "alignItems": "center",
            "flex": 0.95,
            "justifyContent": "space-around",
            "marginBottom": 13.34,
            "marginTop": 20.009999999999998,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "height": 133.4,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 67.5,
                "fontWeight": "bold",
                "marginTop": 1.334,
              }
            }
          >
            Common Risk Factors
          </Text>
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 37.5,
                "textAlign": "center",
              }
            }
          >
            Select all that apply:
          </Text>
        </View>
        <View
          style={
            Object {
              "alignContent": "space-between",
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "space-around",
              "width": "100%",
            }
          }
        >
          <ActivityIndicator
            animating={true}
            color="#004EFF"
            hidesWhenStopped={true}
            size="large"
          />
        </View>
      </View>
      <Text />
    </View>
  </ViewManagerAdapter_ExpoLinearGradient>
</View>
`;
